components/Calendar/Calendar.jsx

```diff
@@ -1,10 +1,166 @@
+
+  const [selectedDate, setselectedDate] = React.useState(stateStart);
+  const [exercises, setExercises] = React.useState([]);
+  const [markedDates, setMarkedDates] = React.useState(setDates(pastWorkoutDates));
+  const [showWorkout, setShowWorkout] = React.useState(false);
+  const firstRun = React.useRef(true);
+
+  const db = firebase.firestore();
+  const currentUser = firebase.auth().currentUser.uid;
+  const userRef = db.collection('users').doc(currentUser);
+
+  const dispatch = useDispatch();
+
+  const filterRecords = (date, records) => records.filter((e) => e.date === date);
+
+  const getDateRecords = async (user, currentDate) => {
+    const recordCheck = filterRecords(currentDate, workoutRecords);
+    if (recordCheck.length > 0) {
+      console.log('Record found in redux.');
+      setShowWorkout(true);
+      setExercises(recordCheck);
+    } else {
+      const recordsRef = user.collection('workoutRecords');
+      const workoutRecordSnapshot = await recordsRef.where('date', '==', currentDate).get();
+
+      if (workoutRecordSnapshot.empty) {
+        setShowWorkout(false);
+        return;
+      }
+      setShowWorkout(true);
+
+      workoutRecordSnapshot.forEach((doc) => {
+        const data = doc.data();
+        const exerciseRecord = data.exercises;
+        const { date } = data;
+
+        const workoutSets = [];
+
+        exerciseRecord.forEach((record) => {
+          const loggableData = {
+            name: record.exerciseName,
+            sets: record.sets,
+            isReps: true,
+            date,
+          };
+          workoutSets.push(loggableData);
+          dispatch(actions.records.addWorkoutRecord(loggableData));
+        });
+        setExercises(workoutSets);
+      });
+    }
+  };
+
+  return (
+    <View style={{ justifyContent: 'center', height: '100%' }}>
+      <Calendar
+        style={{
+          marginTop: 50,
+        }}
+        theme={{
+          backgroundColor: '#f2f2f2',
+          calendarBackground: '#f2f2f2',
+          todayTextColor: '#cab0ff',
+          dotColor: '#cab0ff',
+          arrowColor: '#cab0ff',
+          monthTextColor: 'black',
+          textDayFontFamily: 'Montserrat_500Medium',
+          textMonthFontFamily: 'Montserrat_600SemiBold',
+          textDayHeaderFontFamily: 'Montserrat_500Medium',
+          textDayFontSize: 14,
+        }}
+        markedDates={markedDates}
+        onDayPress={(date) => {
+          const allDates = setDates(pastWorkoutDates);
+          if (allDates[date.dateString]) {
+            allDates[date.dateString] = { selected: true, marked: allDates[date.dateString].marked, selectedColor: '#cab0ff' };
+          } else {
+            allDates[date.dateString] = { selected: true, marked: false, selectedColor: '#cab0ff' };
+          }
+          setMarkedDates(allDates);
+
+          if (firstRun.current) {
+            firstRun.current = false;
+          }
```

Comment:
What is firstRun used for?

Reply:
when the calendar component first loads up it says "no date selected". I think we were planning on changing that to having the current day automatically selected but forgot to implement it

Reply:
Oh we would also still probably need it if we wanted to have today's date selected when it loads up